packages/client/modules/userDashboard/components/OrgBilling/NewBillingLeaderMenu.tsx

```diff
@@ -0,0 +1,118 @@
+import SetOrgUserRoleMutation from '../../../../mutations/SetOrgUserRoleMutation'
+import useAtmosphere from '../../../../hooks/useAtmosphere'
+import useMutationProps from '../../../../hooks/useMutationProps'
+
+const AvatarBlock = styled('div')({
+  paddingRight: 32
+})
+
+interface Props {
+  menuProps: MenuProps
+  organizationRef: NewBillingLeaderMenu_organization$key
+  newLeaderSearchQuery: string
+}
+
+const getValue = (
+  orgUser: NewBillingLeaderMenu_organization$data['organizationUsers']['edges'][0]
+) => orgUser.node.user.preferredName
+
+const NewBillingLeaderMenu = forwardRef((props: Props, ref: any) => {
+  const {menuProps, organizationRef, newLeaderSearchQuery} = props
+  const atmosphere = useAtmosphere()
+  const {onError, onCompleted} = useMutationProps()
+  const organization = useFragment(
+    graphql`
+      fragment NewBillingLeaderMenu_organization on Organization {
+        id
+        billingLeaders {
+          id
+        }
+        organizationUsers {
+          edges {
+            node {
+              id
+              user {
+                id
+                preferredName
+                picture
+              }
+            }
+          }
+        }
+      }
+    `,
+    organizationRef
+  )
+  const {id: orgId, organizationUsers, billingLeaders} = organization
+  const nonLeaderOrgUsers = useMemo(() => {
+    return organizationUsers.edges.filter((organizationUser) => {
+      const {node} = organizationUser
+      const {user} = node
+      const {id: userId} = user
+      return !billingLeaders.some((billingLeader) => {
+        const {id: billingLeaderId} = billingLeader
+        return billingLeaderId === userId
+      })
+    })
+  }, [billingLeaders, organizationUsers])
+
+  const query = newLeaderSearchQuery.toLowerCase()
+  const filteredOrgUsers = useFilteredItems(query, nonLeaderOrgUsers, (orgUser) =>
+    getValue(orgUser).toLowerCase()
+  )
+
+  const handleClick = (userId: string) => {
+    const role = 'BILLING_LEADER'
+    const variables = {orgId, userId, role}
+    SetOrgUserRoleMutation(atmosphere, variables, {onError, onCompleted})
+  }
+
+  return (
+    <Menu ariaLabel='Select New Billing Leader' keepParentFocus {...menuProps}>
+      {filteredOrgUsers.length === 0 && (
+        <EmptyDropdownMenuItemLabel key='no-results'>
+          No team members found!
+        </EmptyDropdownMenuItemLabel>
+      )}
+      {filteredOrgUsers.slice(0, 10).map((organizationUser) => {
+        const {node} = organizationUser
+        const {user} = node
+        const {id: userId, preferredName, picture} = user
+        return (
+          <MenuItem
+            ref={ref}
+            key={userId}
+            label={
+              <MenuItemLabel>
+                <AvatarBlock>
+                  <Avatar picture={picture} size={32} />
+                </AvatarBlock>
+                <TypeAheadLabel query={newLeaderSearchQuery} label={preferredName} />
+                {preferredName}
```

Comment:
-1 the name is shown twice
![Screenshot from 2023-05-15 14-59-07](https://github.com/ParabolInc/parabol/assets/7331043/13fc4edf-837e-42e3-bc74-5985e74a1481)

Reply:
Fixed!